### ISSUE TITLE
Spacing for melismas prior to EOL custos

### ISSUE THREAD

**Issue by Alexander-Eager** [gregorio-project/gregorio#1369]
tl;dr: create a second spacebeforeeolcustos length for line breaks which occur in the middle of a melisma.

Long version:

I looked through the documentation and couldn't find a solution to this problem, and when I looked through the issue log it didn't appear under any search for "custo" or "custos", so I decided to post this.

I'm attempting to mimic the spacing of the Vatican Edition, and part of that spacing is that if a melisma is split across multiple lines, the spacing and rhythm is determined by the distance between the last note on the line and the custos.  So for example, in the Alleluia for the VII Sunday after Pentecost:

<img width="455" alt="alleluia-vii-sunday-after-pentecost" src="https://user-images.githubusercontent.com/6404373/27520385-8a489b8a-59bf-11e7-93af-a6815975bc3d.png">

On the first line, the jubilus continues onto the next line, and the clivis prior to the line break should be lengthened, based on the space between the second note of the clivis and the custos.  On the break between the second and third lines, however, there is a change of syllables, so the spacing can vary without having any rhythmic indication.

Here is an instance where there is no space between the last note of the line and the custos, so nothing should be lengthened:

<img width="455" alt="tract-after-fifth-reading-ember-saturday-in-lent" src="https://user-images.githubusercontent.com/6404373/27520553-e2389e3c-59c2-11e7-96d8-60ca5079fd5c.png">

One solution that I tried was removing all space before the custos by setting spacebeforeeolcustos to be the same as interglyphspace.  This worked perfectly for the melismatic line breaks (any spacing placed after a neume was placed between that neume and the custos) but that also makes inter-syllable line breaks tight as well, and the result is less than satisfying.  

My proposed solution is this: create two new lengths -- something like spacebeforeeolcustosmelisma and spacebeforeeolcustosbetweensyllables -- and link them by default to inherit their value from spacebeforeeolcustos, but allow them to be separated.  That way the current behavior for inter-syllable line breaks can be mixed with Vatican-Edition-friendly melismatic line break formatting.

example.tex

```
\documentclass[11pt,a4paper]{article}

\usepackage{fullpage}

\usepackage{fontspec}
\usepackage{libertine}

\usepackage[autocompile]{gregoriotex}

% this is the length that should be added for melismatic line breaks
% \grechangedim{spacebeforeeolcustosmelisma}{0.06927 cm plus 0.00363 cm minus 0.00363 cm}{scalable}

% alternatively, this could be enabled/disabled with a switch

\begin{document}

\gregorioscore{example}

\end{document}
```

example.gabc

```
name: Example;
%%
(c3) Melisma1(fghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghZ)

Melisma2(fghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfghfgh fghfgh)
```

In this example, when compiled with Gregorio 4.2.0, you get too much space before the custos.  The second melisma is correctly rendered.

**Comment by eroux** [comment]
I think it's a good idea and it should be easy enough... I think having so much space at the end of the line is really terrible and is clearly a mistake, but that's not very important, we can certainly provide this feature!

**Comment by Alexander-Eager** [comment]
Awesome, thank you.  Let me know if you need anything.
